app/models/user_annotation.rb

```diff
@@ -2,24 +2,47 @@ class UserAnnotation
   include Mongoid::Document
   field :name, type: String
   field :values, type: Array
+  field :queued_for_deletion, type: Boolean, default: false
 
   belongs_to :user
   belongs_to :cluster_group
   belongs_to :study
 
-  #user data arrays belong to user annotations
+  # user data arrays belong to user annotations
   has_many :user_data_arrays do
     def by_name_and_type(name, type, subsample_threshold=nil, subsample_annotation=nil)
       where(name: name, array_type: type, subsample_threshold: subsample_threshold, subsample_annotation: subsample_annotation).order_by(&:array_index).to_a
     end
   end
 
+  has_many :user_annotation_shares, dependent: :delete do
```

Comment:
Is there a reason you're only returning the emails in these association queries?  Traditionally you'd return the entire object, not just an attribute.